spec/unit/module_loader/puppetfile_spec.rb

```diff
@@ -99,6 +100,12 @@
       expect(subject.modules.collect(&:name)).not_to include('test_module')
     end
 
+    it 'should not modify the overrides when adding modules' do
+      module_opts = { git: 'git@example.com:puppet/test_module.git' }
+      subject.add_module('puppet/test_module', module_opts)
+      expect(subject.instance_variable_get("@overrides")[:modules]).to eq({exclude_spec: true})
```

Comment:
So this would have come out false before? Is that because we're passing a reference to the `overrides` hash rather than making a copy when we take it into the `module/base` class?

Reply:
Not exactly; since the `base` module class deleted the key, the entire entry in the hash would be missing; if you run the test with the old code, you'll see this error message:

```
r10k: be rspec spec/unit/module_loader/puppetfile_spec.rb
............F............................

Failures:

  1) R10K::ModuleLoader::Puppetfile adding modules should not modify the overrides when adding modules
     Failure/Error: expect(subject.instance_variable_get("@overrides")[:modules]).to eq({exclude_spec: true})

       expected: {:exclude_spec=>true}
            got: {}

       (compared using ==)

       Diff:
       @@ -1,2 +1 @@
       -:exclude_spec => true,

     # ./spec/unit/module_loader/puppetfile_spec.rb:106:in `block (3 levels) in <top (required)>'

Finished in 0.07463 seconds (files took 0.46108 seconds to load)
41 examples, 1 failure
```

This means that the `exclude_spec` would end up as `false` for all remaining base class initializations.

Reply:
Huh okay. Wild.